wp-youtube-live.php

```diff
@@ -188,7 +188,7 @@ function get_youtube_live_content( $request_options ) {
 	}
 
 	// debugging.
-	if ( get_option( 'youtube_live_settings', 'debugging' ) && is_user_logged_in() ) {
+	if ( 'false' !== $youtube_options['debugging'] && is_user_logged_in() ) {
```

Comment:
How about this rather than checking for the string `'false'`?

```suggestion
	if ( $youtube_options['debugging'] && is_user_logged_in() ) {
```

Reply:
Thanks much for your quick review! Re: your suggestion, in fact, that's exactly what I did at the beginning, and it didn't work, so it kept me scratching my head. I eventually delved into the `wp_options` table (whatever the table name is), and saw the content of the options. Not sure the exact format (didn't look like json-encoding), but I figured the setting value is not a boolean but a string. And a non-empty string `'false'` is actually a boolean true, and it made sense. I also tried `boolval()` after looking up the web for string->boolean conversion, but of course that applies to an empty string as false, not the literal `'false'` to false. That's what made me to try this string comparison (I know it's ugly and I don't like it), and it worked. I suspect a proper fix would be to set the correct type for the setting, but that's beyond me, and I wouldn't want to blow up this PR that much, so this quick/small fix.

Just FYI, https://www.insightmeditationcenter.org/2024/01/hs-test-youtube-live-plugin/ is where I installed your plugin, applies this fix, and used the short code in the post. Without this fix, it kept showing the red error message even with the plugin's debugging setting set to false. Thanks again for the great plugin!

Reply:
> Not sure the exact format (didn't look like json-encoding)

No, WP uses PHP’s [`serialize()` function](https://www.php.net/manual/en/function.serialize.php) for options, meta etc.

Ah, it’s storing true/false as strings because they’re strings [here](https://github.com/macbookandrew/wp-youtube-live/blob/9a35b2986e2f0b79fc340857e175a66075f59b7d/inc/admin.php#L307).

How about we check for the positive condition rather than absence of negative condition?

```suggestion
	if ( 'true' === $youtube_options['debugging'] && is_user_logged_in() ) {
```